Adafruit_MQTT.cpp

```diff
@@ -233,7 +233,7 @@ uint16_t Adafruit_MQTT::readFullPacket(uint8_t *buffer, uint16_t maxsize,
   // will read a packet and Do The Right Thing with length
   uint8_t *pbuff = buffer;
 
-  uint8_t rlen;
+  uint16_t rlen;
```

Comment:
Packets larger than 255 bytes will overflow rlen.

Reply:
This does solve an issue I was having today with a very large binary payload, thank you.

> Correctly identify topic start position for >127byte messages (may also be addressed in PR166 -- happy to merge that fix in if preferred).

@Fapiko  Is this implemented in #166?

Reply:
hi @brentru . No, this fix is related but not exactly what I was trying to address in https://github.com/adafruit/Adafruit_MQTT_Library/pull/166 . The https://github.com/adafruit/Adafruit_MQTT_Library/pull/166 was done to avoid corrupting memory should we attempt to
publish more data than what is available by MAXBUFFERSIZE. I think the function introduced in this PR (topicOffsetFromLenth) is what https://github.com/adafruit/Adafruit_MQTT_Library/pull/166 introduces, so we definitely should combine that. @xdylanm how about you incorporate that into this pr? You can take the credit; I do not care for that. ;) Basically, we need this logic in addition to what you are doing here: https://github.com/adafruit/Adafruit_MQTT_Library/pull/166/commits/d6f07ed4e4b768a62ee5510aa19b595b1b8dc42f#diff-a20445c19076bfb61fa2cf57b03325851600fc2dc25391f7e97f2de072fda081R701

Reply:
@flavio-fernandes in my own build, I originally had a simpler version of the length test 
`uint16_t const topicoffset = (len > 127 ? 1 : 0)`
but I liked your function approach better, so I followed that pattern and credit to you =). If #166 is close to being merged, I'm happy to just rebase off of that when it's merged and update my PR. Per your note, you're trying to solve a different problem, which should certainly get resolved. 

That said, I realized that this solution does not support packets > 65535 bytes since the lengths are all stored in uint16_t, and I'm still not sure what the right solution is there.

Reply:
Understood. Yes, you correctly point out the fact that we cannot go beyond 65535 since we are using uint16.

I will defer to @brentru on how we want to proceed here, but my opinion is to go ahead with PR/166 as is,
and then have you leverage the function *packetAdditionalLen* in PR/193 and potentially improve on the uint16 limitation. Sounds reasonable, @brentru ?

Reply:
This sounds OK and I agree. - @xdylanm, see the comment below about using clang-format so we can approve this PR.